arn/main.go

```diff
@@ -32,8 +32,16 @@ func ParseP(arn *string) Arn {
 	return Parse(*arn)
 }
 
+func InjectResourceIdAndVersion(arn Arn, resourceArnPart string) Arn {
```

Comment:
This function isn't needed outside of the `arn` package, so it should be lower case